README.dev.md

```diff
@@ -449,3 +449,24 @@ Possible testing issues
 
 - When building the development environment using Vagrant and the LXC provider, if you encounter permissions errors, you may need to `chown -R 1000:$(whoami) /path/to/zulip` on the host before running `vagrant up` in order to ensure that the synced directory has the correct owner during provision. This issue will arise if you run `id username` on the host where `username` is the user running Vagrant and the output is anything but 1000.
   This seems to be caused by Vagrant behavior; more information can be found [here](https://github.com/fgrehm/vagrant-lxc/wiki/FAQ#help-my-shared-folders-have-the-wrong-owner).
+
+Troubleshooting development environment setup
+=============================================
+If you are building the development environment behind the network proxy then make sure your proxy variables are set. 
+
+- On ubuntu, Set the proxy environment variable using 
+ ``` 
+     export https_proxy=https://proxy_host:port 
+     export http_proxy=https://proxy_host:port 
+ ```
+
+- And set the npm proxy and https-proxy separately as
+ ``` 
+     npm config set proxy http://proxy_host:port
+     npm config set https-proxy http://proxy_host:port
+ ```
+
+- Also add a blank .npmignore file into the main zulip directory
+  ```
+     touch .npmignore
```

Comment:
Ok sounds like we should add an nomignore file then

-Tim Abbott (mobile)
On Mar 19, 2016 1:21 PM, "Ashish Kumar" <notifications@github.com> wrote:

> In README.dev.md
> <https://github.com/zulip/zulip/pull/539#discussion_r56754521>:
>
> > +
> > +- On ubuntu, Set the proxy environment variable using
> > + ```
> > +     export https_proxy=https://proxy_host:port
> > +     export http_proxy=https://proxy_host:port
> > + ```
> > +
> > +- And set the npm proxy and https-proxy separately as
> > + ```
> > +     npm config set proxy http://proxy_host:port
> > +     npm config set https-proxy http://proxy_host:port
> > + ```
> > +
> > +- Also add a blank .npmignore file into the main zulip directory
> > +  ```
> > +     touch .npmignore
>
> @timabbott <https://github.com/timabbott> I have tested it in the above
> three ways in my machine but every time by creating .npmignore or deleting
> node_modules from .gitignore it passes the installation process. I am also
> not sure whether it is a npm transient issue or not (maybe because of my
> machine specifications... 32 bit 12.04 ubuntu).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zulip/zulip/pull/539/files/abc7f52945961657235f42a5cd6aad8a8eed5871#r56754521>
>